spikeinterface/toolkit/preprocessing/common_reference.py

```diff
@@ -0,0 +1,153 @@
+        If 'single', the selected channel(s) is remove from all channels.
+        If 'local', an average CAR is implemented with only k channels selected the nearest outside of a radius around each channel
+    groups: list
+        List of lists containing the channel ids for splitting the reference. The CMR, CAR, or referencing with respect to
+        single channels are applied group-wise. However, this is not applied for the local CAR.
+        It is useful when dealing with different channel groups, e.g. multiple tetrodes.
+    ref_channels: list or int
+        If no 'groups' are specified, all channels are referenced to 'ref_channels'. If 'groups' is provided, then a
+        list of channels to be applied to each group is expected. If 'single' reference, a list of one channel  or an
+        int is expected.
+    local_radius: tuple(int, int)
+        Use in the local CAR implementation as the selecting annulus (exclude radius, include radius)
+    dtype: str
+        dtype of the returned traces. If None, dtype is maintained
+    verbose: bool
+        If True, output is verbose
+
+    Returns
+    -------
+    referenced_recording: CommonReferenceRecording
+        The re-referenced recording extractor object
+    """
+    
+    name = 'common_reference'
+
+    def __init__(self, recording, reference='median', groups=None, ref_channels=None,
+        local_radius=(2, 8), verbose=False):
+
+        self._kwargs = dict(recording=recording.to_dict(), reference=reference, groups=groups,
+                ref_channels=ref_channels, local_radius=local_radius)
+        
+        num_chans = recording.get_num_channels()
+        neighbors = None
+        # some checks
+        if reference not in ('median', 'average', 'single', 'local'):
+            raise ValueError("'reference' must be either 'median', 'average', 'single' or 'local'")
+
+        if reference == 'single':
+            assert ref_channels is not None, "With 'single' reference, provide 'ref_channels'"
+            if groups is not None:
+                assert len(ref_channels) == len(_groups), \
+                        "'ref_channel' and 'groups' must have the same length"
+            else:
+                if np.isscalar(ref_channels):
+                    ref_channels = [ref_channels]
+                else:
+                    assert len(ref_channels) == 1, \
+                        "'ref_channel' with no 'groups' can be int or a list of one element"
+                ref_channels = np.asarray(ref_channels)
+                assert ref_channels.dtype.kind == 'i',"'ref_channels' must be int"
+        elif reference == 'local':
+            assert groups is None, "With 'local' CAR, the group option should not be used."
+            closest_inds, dist = get_closest_channels(recording)
+            
+            neighbors = {}
+            for i in range(num_chans):
+                mask = (dist[i, :] > local_radius[0]) & (dist[i, :] < local_radius[1])
```

Comment:
I would put `mask = (dist[i, :] > local_radius[0]) & (dist[i, :] <= local_radius[1])`
If both circle are very close,  it would make sense for me that channels on the upper bounds of the annulus is still count in it. 
I guess it is only a matter of implementation. However, if you prefer to keep a strict upper bound, you should also change it in the actual version of spiketoolkit to keep the same behavior over your different versions.

Reply:
ok I will do.
I miss that in your implementation.
I think I would also do it for low limit.